torchvision/transforms/transforms.py

```diff
@@ -420,15 +466,13 @@ class RandomApply(RandomTransforms):
     """
 
     def __init__(self, transforms, p=0.5):
-        super(RandomApply, self).__init__(transforms)
+        super().__init__(transforms)
         self.p = p
 
-    def __call__(self, img):
-        if self.p < random.random():
+    def forward(self, img: Tensor) -> Tensor:
+        if self.p < torch.rand(1):
             return img
-        for t in self.transforms:
-            img = t(img)
-        return img
+        return self._forward_impl(img)
```

Comment:
Now I see why you implemented `Compose` with `_forward_impl`. I wonder if this is really necessary, given that we only save two lines but it adds additional complexity to the user to read the function?

Reply:
Personnally, I'd prefer to refactor... If you think that it is better for transparency reasons to put explicit code, I'm fine with that too.